### PR TITLE
feat: ng-add schematics implementation

### LINKED ISSUE #9: [Feat] Add ng add support
- Updates the package.json
- Adds RoutingModule.forRoot() to the AppModule

### PR DESCRIPTION
**Pull request by santoshyadavdev**
fixes #9

### CODE REVIEW

**Comment by meeroslav** [comment]
Hi @santoshyadavdev, 
please rename the PR to mark clearly what it's about.
Something like: `build: add ng add support. resolves #9`.

Thanks!

**Comment by santoshyadavdev** [comment]
Hey yeah will do it in next commit it was just to verify the schematic project structure 😊

**Comment by brandonroberts** [comment]
@santoshyadavdev the schematic project structure is good

**Comment by santoshyadavdev** [comment]
Hi @meeroslav please wait for review this is draft PR.😊

**Comment by santoshyadavdev** [comment]
Hi @brandonroberts ,
Just wanted to add the test config, will wrap tomorrow.

File: libs/angular-routing/package.json [hunk 1]
```diff
@@ -2,6 +2,10 @@
   "name": "angular-routing",
   "version": "0.1.1",
   "description": "A declarative router for Angular applications",
+  "schematics": "./schematics/collection.json",
+  "ng-add": {
```
Comment (brandonroberts) [inline libs/angular-routing/package.json:6]: Nice! I didn't know you could do this.

File: package.json [hunk 2]
```diff
@@ -25,7 +25,12 @@
     "update": "ng update @nrwl/workspace",
     "workspace-schematic": "nx workspace-schematic",
     "dep-graph": "nx dep-graph",
-    "help": "nx help"
+    "help": "nx help",
+    "build:schematics": "tsc -p tsconfig.schematics.json",
+    "copy:schemas": "copyfiles  schematics/*/schema.json dist/libs/angular-routing/",
```
Comment (brandonroberts) [inline package.json:30]: We don't need an external package for this. ng-packagr has an `assets` property in the `ng-package.json` to add globs for files.

File: package.json [hunk 3]
```diff
@@ -25,7 +25,12 @@
...
+    "copy:collection": "copyfiles schematics/collection.json dist/libs/angular-routing/",
+    "schematics:postbuild": "npm run copy:schemas && npm run copy:collection",
+    "schematics:build": "npm run build:schematics && npm run schematics:postbuild"
```
Comment (brandonroberts) [inline package.json:33]: Instead let's define a new target in the angular.json, such as `build-package` to build the lib, and use `run-commands` builder to run `build-package`, and compile the schematics. See

https://github.com/ngrx/platform/blob/master/angular.json#L236

File: schematics/ng-add/schema.json [hunk 4]
```diff
@@ -0,0 +1,28 @@
+  {
+    "$schema": "http://json-schema.org/schema",
+    "id": "SchematicsNgRxRootEffect",
+    "title": "NgRx Root Effect Options Schema",
```
Comment (brandonroberts) [inline schematics/ng-add/schema.json:4]:
```suggestion
    "id": "SchematicsAddRouting",
    "title": "Angular Routing ng-add Options Schema",
```

File: tsconfig.schematics.json [hunk 5]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (brandonroberts) [inline tsconfig.schematics.json:1]: This file should be in the `libs/angular-routing` folder

File: package.json (hunk 3 above)
Comment (santoshyadavdev) [reply package.json:33]: This is nice.

File: libs/angular-routing/package.json (hunk 1 above)
Comment (santoshyadavdev) [reply libs/angular-routing/package.json:6]: Yeah Angular CLI ng add command has this built-in now.

File: package.json (hunk 3 above)
Comment (santoshyadavdev) [reply package.json]: Removed.

File: schematics/ng-add/schema.json (hunk 4 above)
Comment (santoshyadavdev) [reply schematics/ng-add/schema.json]: Done.

File: tsconfig.schematics.json (hunk 5 above)
Comment (santoshyadavdev) [reply tsconfig.schematics.json]: Done.

File: libs/angular-routing/schematics/ng-add/index.ts [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    ];
+
+    const recorder = host.beginUpdate(modulePath);
+
+    for (const change of changes) {
+      if (change instanceof InsertChange) {
+        recorder.insertLeft(change.pos, change.toAdd);
+      }
+    }
+    host.commitUpdate(recorder);
```
Comment (timdeschryver) [inline libs/angular-routing/schematics/ng-add/index.ts:59]: Fyi, we have a helper function for this 🙂
```suggestion
    commitChanges(host, modulePath, changes);
```

File: libs/angular-routing/schematics/ng-add/schema.ts [hunk 7]
```diff
@@ -0,0 +1,7 @@
+export interface RouterOptions {
+  project: string;
+  module?: string;
+  name: string;
```
Comment (timdeschryver) [inline libs/angular-routing/schematics/ng-add/schema.ts:4]: I think `name` can be removed, as it isn't added to `schema.json`
```suggestion
```

File: libs/angular-routing/schematics/ng-add/schema.ts [hunk 8]
```diff
@@ -0,0 +1,7 @@
...
+  path?: string;
+  skipImport?: boolean;
```
Comment (timdeschryver) [inline libs/angular-routing/schematics/ng-add/schema.ts:6]: Same here
```suggestion
```

**Comment by santoshyadavdev** [comment]
Your review is always helpful @timdeschryver 💙 thanks for taking your time 🤗

File: libs/angular-routing/schematics/ng-add/schema.ts (hunk 7 above)
Comment (santoshyadavdev) [reply libs/angular-routing/schematics/ng-add/schema.ts:4]: > I think `name` can be removed, as it isn't added to `schema.json`

It is used internally by findModuleFromOptions

**Comment by santoshyadavdev** [comment]
Hi @brandonroberts , resolved all conflicts, let me know if you want to squash commits.

**Comment by brandonroberts** [comment]
Thanks @santoshyadavdev!

**Pull request merged**